### PR TITLE
fix(ecma/transform/compat): transform new.target in arrow

### LINKED ISSUE #2885: `new.target` in arrow function wrongly transformed
### Describe the bug

This is similar to #2811, the problem lies in chrome 46 support `new.target` but not `arrow`

### Input code

```typescript
function foo() {
  const a = (a) => new.target
}
```

### Config

```js
{
  jsc: {
    parser: {
      syntax: "ecmascript"
    }
  },
  env: {
    targets: {
      chrome: "46"
    },
    mode: "entry",
    coreJs: 3
  }
}
```

### Playground link

_No response_

### Expected behavior

should output
```js
function foo() {
  var _newtarget = new.target;

  var a = function a(_a) {
    return _newtarget;
  };
}
```

### Version

1.2.112

### Additional context

There are two solutions.

1. Remove arrow related transform in `new.target` and move it to `arrow`
   pros: it's clean and can be reused in other transforms
   cons: it will need `arrow` transform be placed before `new.target`
2. Make `new.target` aware arrow function don't have its own `new.target` and reuse it in other transforms
   cons: it's ugly

### PR DESCRIPTION
**Pull request by Austaras**
**Related issue (if exists):**

closes #2885 

Also why `new_target` transform not in `preset_env`?

### CODE REVIEW

**Comment by kdy1** [comment]
Can you add it?
There's no good way to ensure that we used all compat transforms in preset-env

**Comment by Austaras** [comment]
Added. I'm just making sure there isn't a specific reason not to

File: crates/swc_ecma_utils/src/function.rs [hunk 1]
```diff
@@ -7,14 +7,15 @@ use swc_ecma_visit::{noop_visit_mut_type, VisitMut, VisitMutWith};
 pub struct WrapperState {
     this: Option<Ident>,
     args: Option<Ident>,
+    target: Option<Ident>,
```
Comment (kdy1) [inline crates/swc_ecma_utils/src/function.rs:10]: Can you add a comment that this `target` is `new.target`?

File: crates/swc_ecma_utils/src/function.rs [hunk 2]
```diff
@@ -7,14 +7,15 @@ use swc_ecma_visit::{noop_visit_mut_type, VisitMut, VisitMutWith};
...
-    pub fn new(this: Option<Ident>, args: Option<Ident>) -> Self {
-        Self { this, args }
+    pub fn new(this: Option<Ident>, args: Option<Ident>, target: Option<Ident>) -> Self {
+        Self { this, args, target }
     }
     pub fn to_decl(self) -> Vec<VarDeclarator> {
-        let Self { this, args } = self;
+        let Self { this, args, target } = self;
 
         let mut decls = Vec::with_capacity(2);
```
Comment (kdy1) [inline crates/swc_ecma_utils/src/function.rs:20]: This can be 3 I guess?

File: crates/swc_ecma_utils/src/function.rs [hunk 3]
```diff
@@ -54,7 +66,7 @@ impl WrapperState {
     }
 
     pub fn to_stmt_in_subclass(self) -> (Option<Stmt>, Option<Ident>) {
-        let Self { this, args } = self;
+        let Self { this, args, target } = self;
 
         let mut decls = Vec::with_capacity(2);
```
Comment (kdy1) [inline crates/swc_ecma_utils/src/function.rs:71]: This can be 3 I guess?

**Review by kdy1 — CHANGES_REQUESTED** [review]
Almost LGTM. Thank you!

**Comment by Austaras** [comment]
Done

**Review by kdy1 — APPROVED** [review]
LGTM. Thanks!

**Pull request merged**